qiskit/result/utils.py

```diff
@@ -80,7 +80,7 @@ def marginal_counts(
                     )  # same convention as for the counts
                     bit_strings = [_hex_to_bin(s) for s in experiment_result.data.memory]
                     marginal_bit_strings = [
-                        "".join([s[-idx - 1] for idx in sorted_indices if idx < len(s)])
+                        "".join([s[-idx - 1] for idx in sorted_indices if idx < len(s)]) or "0"
```

Comment:
Can the empty string condition only be triggered in case of a `0`? I'm just thinking the same thing could come up if there are different bit string lengths and the marginalization might not always be zero. Like if you say I I'm interested in bit index `1` and `3` and you have a bit string `1` instead of `0001` this will make the marginalized string `0` instead of `00`. Maybe we need to start padding the bitstrings to a fixed width here?

Reply:
Oh yeah, that's a good point.  It could be `"0" * len(sorted_indices)`.

Reply:
No wait, I don't think it matters.  This is just the memory slots, which are represented by a hex integer, so `"00"` is the same as `"0"` - they both become `"0x0"` right after.

Reply:
Ah right, I forgot that this function converts back to hex in the return. I have a commit/branch for a pending PR built on top of #8026 adding memory marginalization functions (https://github.com/mtreinish/qiskit-core/commit/3298f2271ec83a3b9c905604e5e2530bcaf263a9 ) and this caught me by surprise there too.